### PR TITLE
Me testing

### PR DESCRIPTION
**Pull request by Sphwen**
This PR implements logging and testing methods for the MatchingEngine, as well as changes to the ME structuring and functionality. I intend to put up another PR to make better use of the restructuring of order handling, but this should at least get log messages up for anyone who needs them.

### CODE REVIEW

File: matching-engine/src/main/java/matchingengine/MatchingEngine.java [hunk 1]
```diff
@@ -53,224 +69,147 @@ public void register(){
...
-                cancelledQuantity=buyers.get(i).getQuantity();
-                buyers.remove(i);
+        for(int i=0;i<restingBook.buyers.size()&&done==false;i++){
+            if(restingBook.buyers.get(i).getPlayerId()==playerId&&restingBook.buyers.get(i).getOrderID()==orderId){
+                cancelledQuantity=restingBook.buyers.get(i).getQuantity();
+                restingBook.buyers.remove(i);
                 done=true;
             }
         }
         //TODO what message do we send if the order does not exist? or how do we format this message
```
Comment (TannerKvarfordt) [inline matching-engine/src/main/java/matchingengine/MatchingEngine.java]: I imagine just send back a CancelConfirmationMessage with a cancelledQuantity of 0.

File: matching-engine/src/main/java/matchingengine/OrderHandler.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            for(int i=0;i<restingBook.sellers.size()&&!orderComplete;i++){
+                if(orderPrice>=restingBook.sellers.get(i).getPrice()){
+                    if(restingBook.sellers.get(i).getQuantity()==remainingOrderQuantity){
+                        orderComplete=true;
+                        OrderConfirmationMessage orderConfirmationMessage1=new OrderConfirmationMessage(
+                                restingBook.sellers.get(i).getPlayerId(),restingBook.sellers.get(i).getOrderID(),
+                                restingBook.sellers.get(i).getQuantity(), (short)0,restingBook.sellers.get(i).getPrice(),
+                                restingBook.sellers.get(i).getSymbol()
+                        );
+                        //TODO should this orderprice be sellers(i) order price?
```
Comment (TannerKvarfordt) [inline matching-engine/src/main/java/matchingengine/OrderHandler.java]: I think it should be sellers(i) order price. That way the buyer gets the best price on offer, and the seller gets what they ask for. Can @CarsonBradshaw or @Palexander21 verify?

Comment (Sphwen) [reply matching-engine/src/main/java/matchingengine/OrderHandler.java]: Yeah i know you're right on this one we talked about it but I totally

Comment (Sphwen) [reply matching-engine/src/main/java/matchingengine/OrderHandler.java]: sorry lol. Anyway I spaced making the change but I'll correct it

Comment (TannerKvarfordt) [reply matching-engine/src/main/java/matchingengine/OrderHandler.java]: No worries! There's a lot to keep track of.

File: matching-engine/src/main/java/matchingengine/MatchingEngine.java (hunk 1 above)
Comment (Sphwen) [reply matching-engine/src/main/java/matchingengine/MatchingEngine.java]: yes you're right.

**Pull request merged**